src/pages/ServiceList/ServiceListComponent.tsx

```diff
@@ -178,208 +79,96 @@ class ServiceListComponent extends React.Component<ServiceListComponentProps, Se
-        services: prevState.services,
-        pagination: {
-          page: page,
-          perPage: prevState.pagination.perPage,
-          perPageOptions: ListPage.perPageOptions
-        }
-      };
-    });
-
-    this.props.pageHooks.onParamChange([{ name: 'page', value: String(page) }]);
-  };
-
-  perPageSelect = (perPage: number) => {
-    this.setState(prevState => {
-      return {
-        services: prevState.services,
-        pagination: {
-          page: 1,
-          perPage: perPage,
-          perPageOptions: ListPage.perPageOptions
-        }
-      };
-    });
-
-    this.props.pageHooks.onParamChange([{ name: 'page', value: '1' }, { name: 'perPage', value: String(perPage) }]);
-  };
-
-  updateSortField = (sortField: SortField) => {
-    sortServices(this.state.services, sortField, this.state.isSortAscending).then(sorted => {
-      this.setState({
-        currentSortField: sortField,
-        services: sorted
-      });
-
-      this.props.pageHooks.onParamChange([{ name: 'sort', value: sortField.param }]);
-    });
-  };
-
-  updateSortDirection = () => {
-    sortServices(this.state.services, this.state.currentSortField, !this.state.isSortAscending).then(sorted => {
-      this.setState({
-        isSortAscending: !this.state.isSortAscending,
-        services: sorted
-      });
-
-      this.props.pageHooks.onParamChange([{ name: 'direction', value: this.state.isSortAscending ? 'asc' : 'desc' }]);
-    });
-  };
-
-  updateServices = (resetPagination?: boolean) => {
+  updateListItems(resetPagination?: boolean) {
     const activeFilters: ActiveFilter[] = FilterSelected.getSelected();
     let namespacesSelected: string[] = activeFilters
       .filter(activeFilter => activeFilter.category === 'Namespace')
       .map(activeFilter => activeFilter.value);
-    let servicenameFilters: string[] = activeFilters
-      .filter(activeFilter => activeFilter.category === 'Service Name')
-      .map(activeFilter => activeFilter.value);
-    let istioFilters: string[] = activeFilters
-      .filter(activeFilter => activeFilter.category === 'Istio Sidecar')
-      .map(activeFilter => activeFilter.value);
 
     /** Remove Duplicates */
     namespacesSelected = removeDuplicatesArray(namespacesSelected);
-    servicenameFilters = removeDuplicatesArray(servicenameFilters);
-    istioFilters = this.cleanIstioFilters(removeDuplicatesArray(istioFilters));
 
     if (namespacesSelected.length === 0) {
       API.getNamespaces(authentication())
         .then(namespacesResponse => {
           const namespaces: Namespace[] = namespacesResponse['data'];
           this.fetchServices(
             namespaces.map(namespace => namespace.name),
-            servicenameFilters,
-            istioFilters,
+            activeFilters,
+            this.state.rateInterval,
             resetPagination
           );
         })
         .catch(namespacesError => this.handleAxiosError('Could not fetch namespace list.', namespacesError));
```

Comment:
@lucasponce  @jotak @aljesusg I do have a problem here with the scope of `this` object. It says that is undefined at this point:
`Unhandled Rejection (TypeError): Cannot read property 'handleAxiosError' of undefined`.
Do you have a clue about what is happening here?

Reply:
probably the `handleAxiosError` is defined outside of the class ?